pkgs/games/atlauncher/default.nix

```diff
@@ -0,0 +1,59 @@
+{ copyDesktopItems, fetchurl, jre, lib, makeDesktopItem, makeWrapper, stdenv, steam-run, withSteamRun ? true, writeShellScript }:
+
+stdenv.mkDerivation rec {
+  pname = "atlauncher";
+  version = "3.4.28.1";
+
+  src = fetchurl {
+    url = "https://github.com/ATLauncher/ATLauncher/releases/download/v${version}/ATLauncher-${version}.jar";
+    sha256 = "sha256-IIwDMazxUMQ7nGQk/4VEZicgCmCR4oR8UYtO36pCEq4=";
+  };
+
+  dontUnpack = true;
+
+  buildInputs = [ ];
+  nativeBuildInputs = [ copyDesktopItems makeWrapper ];
+
+  installPhase =
+    let
+      # hack to use steam-run along with the exec
+      steamrun = writeShellScript "steamrun" ''
+        shift
+        exec ${steam-run}/bin/steam-run "''$@"
+      '';
+    in
+    ''
+      runHook preInstall
+      mkdir -p $out/bin
+      makeWrapper ${jre}/bin/java $out/bin/atlauncher \
+        --add-flags "-jar $src --working-dir=\$HOME/.atlauncher" \
+        --suffix LD_LIBRARY_PATH : "${lib.makeLibraryPath buildInputs}" ${
+            if withSteamRun
```

Comment:
There is a lib function for that. I think it's `lib.optionalString` or something like that